lib/spack/spack/cmd/test.py

```diff
@@ -101,26 +131,119 @@ def test(parser, args):
         else:
             log_file = os.path.join(
                 os.getcwd(),
-                'test-%s' % now.strftime('%Y-%m-%d_%H:%M:%S'))
+                'test-%s' % test_name)
         reporter.filename = log_file
     reporter.specs = specs_to_test
 
     # test_stage_dir
-    test_name = now.strftime('%Y-%m-%d_%H:%M:%S')
-    stage = os.path.join(
-        spack.util.path.canonicalize_path(
-            spack.config.get('config:test_stage', os.getcwd())),
-        test_name)
+    stage = _get_stage(test_name)
+    mkdirp(stage)
 
     with reporter('test', stage):
         if args.smoke_test:
             for spec in specs_to_test:
                 try:
                     spec.package.do_test(
-                        time=now,
+                        name=test_name,
                         remove_directory=not args.keep_stage,
                         dirty=args.dirty)
+                    with open(_get_results_file(test_name), 'a') as f:
+                        f.write("%s PASSED\n" % spec.format("{name}-{hash:7}"))
                 except BaseException:
-                    pass  # Test is logged, go on to other tests
+                    with open(_get_results_file(test_name), 'a') as f:
+                        f.write("%s FAILED\n" % spec.format("{name}-{hash:7}"))
         else:
             raise NotImplementedError
+
+
+def test_list(args):
+    """
+    List tests that are running or have available results.
+    """
+    stage_dir = _get_stage()
+    tests = os.listdir(stage_dir)
+
+    # Filter tests by filter argument
+    if args.filter:
+        def create_filter(f):
+            raw = fnmatch.translate('f' if '*' in f or '?' in f
+                                    else '*' + f + '*')
+            return re.compile(raw, flags=re.IGNORECASE)
+        filters = [create_filter(f) for f in args.filter]
+
+        def match(t, f):
+            return f.match(t)
+        tests = [t for t in tests if any(match(t, f) for f in filters)]
+
+    if tests:
+        # TODO: Make these specify results vs active
+        msg = "Spack test results available for the following tests:\n"
+        msg += "        %s\n" % ' '.join(tests)
+        msg += "    Run `spack clean -t` to remove all tests"
+        tty.msg(msg)
+    else:
+        msg = "No test results match the query\n"
+        msg += "        Tests may have been removed using `spack clean -t`"
+        tty.msg(msg)
+
+
+def test_status(args):
+    """
+    Get the current status for a particular Spack test ensemble.
+    """
+    name = args.name
+    stage = _get_stage(name)
+
+    if os.path.exists(stage):
+        # TODO: Make this handle capability tests too
+        tty.msg("Test %s completed" % name)
+    else:
+        tty.msg("Test %s no longer available" % name)
```

Comment:
I am getting this message when I run `spack test status libsigsegv` after the tests complete.

Intuitively, I would expect this to report that the tests actually completed.

Reply:
As with other comments, the `name` here is the one provided (or defaulted) when you run the tests.